### DISCUSSION TITLE
Archiving changes (and viewing them)

### DISCUSSION THREAD

**Discussion by avamsi** [jj-vcs/jj#4180] [Ideas]
I sometimes have these changes that I'm not quite ready to discard but at the same time, I don't want them in my log as well.

I started "tagging" these changes manually with `#jj-archive` (i.e., put this somewhere in the description) and excluding them from my log revset (with `archived = 'description(substring:"#jj-archive")'`) -- this kind of does what I wanted to do and I can easily view my archived changes with `jj log -r archived`.

1. Does anyone else have similar workflows? Do you do something different?
2. Is there a way to simplify the "tagging" part? Ideally this would be a single jj alias called `archive` (I think the --trailer flag from https://github.com/martinvonz/jj/issues/1399#issuecomment-2254168454 would help)?

**Comment by avamsi** [comment]
For posterity, I've since added aliases in [my jjconfig](https://github.com/avamsi/dotfiles/blob/e70df0ad1c58ea9fc831a722b70d43da2efc14c7/.jjconfig.toml#L8) to make this easier (https://github.com/avamsi/dotfiles/issues/10#issuecomment-2262101112).

**Comment by steveklabnik** [comment]
> Is there a way to simplify the "tagging" part?

Sounds like topics, to me, I think? https://github.com/martinvonz/jj/issues/3402

**Reply by avamsi** [reply]
Yep, thanks -- that does seem aligned!

**Comment by alycda** [comment]
For anyone who lands here: I opened #10268, a feature request for a first-class `jj hide` / `jj unhide` covering this use case. It would hide changes without tagging their descriptions, which rewrites them, and would work in tools that don't read `revsets.log`. It links a small prototype and the edge cases it ran into: merges where only one parent is hidden, and ordering between hidden sets.
